pc/protocols/usb_boot.c

```diff
@@ -324,14 +332,34 @@ usbBootError_t usb_find_device_with_bcd(unsigned idx, char *input_addr,
                         fprintf(stderr, "Found Address: %s - VID/PID %04x:%04x\n",
                                 input_addr, desc.idVendor, desc.idProduct);
                     }
-
                     libusb_ref_device(dev);
                     libusb_free_device_list(devs, 1);
                     if (bcdusb)
                         *bcdusb = desc.bcdUSB;
                     *device = dev;
                     devs = 0;
 
+                    int speed = libusb_get_device_speed(dev);
+                    char *speed_str[] = {"Unknown", "Low/1.5Mbps", "Full/12Mbps", "High/480Mbps", "Super/5000Mbps"};
+                    // if(speed != NULL)
+                    //     mv_strcpy(usb_speed, 15 ,speed_str[speed]);
+
+                    libusb_device_handle *dev_handle;
+                    if (libusb_open(dev, &dev_handle) == 0) {
```

Comment:
Is this open required here?

Reply:
It's required to get a handle to pass to `libusb_get_string_descriptor_ascii(dev_handle, desc.iSerialNumber, ...)`, unlike for other API as `libusb_get_device_speed(dev)`.